### PR TITLE
Documents impersonation (#6293)

### PR DESCRIPTION
**Pull request by klizhentas**

### CODE REVIEW

**Review by r0mant — APPROVED** [review]

File: docs/pages/access-controls/guides/impersonation.mdx [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+
+Alice can issue certificates for a security scanner:
+
+``` bash
+$ tctl auth sign --user=security-scanner --format=openssh --out security-scanner --ttl=10h
+```
+
+**Matching user traits**
+
+We can make security-impersonator role more flexible by matching source user and target role parameters:
```
Comment (webvictim) [inline docs/pages/access-controls/guides/impersonation.mdx:233]:
```suggestion
We can make the `security-impersonator` role more flexible by matching source user and target role parameters:
```

File: docs/pages/access-controls/guides/impersonation.mdx [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+spec:
+  traits:
+    group: ['security', 'devops']
+  roles:
+  - security-impersonator
+  - access
+```
+
+```bash
+# Once Alice logins again, she will receive a new certificate with updated roles.
```
Comment (webvictim) [inline docs/pages/access-controls/guides/impersonation.mdx:274]:
```suggestion
# Once Alice logs in again, she will receive a new certificate with updated roles.
```

File: docs/pages/access-controls/guides/impersonation.mdx [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+$ tsh login --proxy=teleport.example.com --user=alice --auth=local
+# Alice can now get a security scanner
```
Comment (webvictim) [inline docs/pages/access-controls/guides/impersonation.mdx:276]:
```suggestion
# Alice can now get a certificate for the security scanner
```

File: docs/pages/access-controls/guides/impersonation.mdx [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+$ tctl --insecure auth sign --user=security-scanner --format=openssh --out security-scanner --ttl=10h
```
Comment (webvictim) [inline docs/pages/access-controls/guides/impersonation.mdx:277]:
```suggestion
$ tctl auth sign --user=security-scanner --format=openssh --out=security-scanner --ttl=10h
```

File: docs/pages/access-controls/guides/impersonation.mdx [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+```
+
+Here is a summary variables and functions we used in this guide:
```
Comment (webvictim) [inline docs/pages/access-controls/guides/impersonation.mdx:280]:
```suggestion
Here is a summary of variables and functions we used in this guide:
```

File: docs/pages/access-controls/guides/impersonation.mdx [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+
+```bash
+$ tctl users add alice  --roles=impersonator,access
+```
+
+Alice can login using `tsh` and issue a cert for `jenkins`:
+
+```bash
+$ tsh login --proxy=teleport.localhost:3080 --user=alice --auth=local
+$ tctl auth sign --user=jenkins --format=openssh --out jenkins --ttl=240h
```
Comment (webvictim) [inline docs/pages/access-controls/guides/impersonation.mdx:119]:
```suggestion
$ tctl auth sign --user=jenkins --format=openssh --out=jenkins --ttl=240h
```

File: docs/pages/access-controls/guides/impersonation.mdx [hunk 7]
```diff
@@ -0,0 +1,299 @@
...
+  labels:
+    group: security
+spec:
+  roles: ['security-scanner']
+```
+
+Alice can issue certificates for a security scanner:
+
+``` bash
+$ tctl auth sign --user=security-scanner --format=openssh --out security-scanner --ttl=10h
```
Comment (webvictim) [inline docs/pages/access-controls/guides/impersonation.mdx:228]:
```suggestion
$ tctl auth sign --user=security-scanner --format=openssh --out=security-scanner --ttl=10h
```

**Review by webvictim — APPROVED** [review]

File: docs/pages/access-controls/guides/impersonation.mdx [hunk 8]
```diff
@@ -0,0 +1,299 @@
...
+Verify that your Teleport client is connected:
+
+```bash
+$ tctl status
+Cluster  acme.example.com
+Version  6.0.2
+Host CA  never updated
+User CA  never updated
+Jwt CA   never updated
+CA pin   sha256:e63c7c44be468d37a5b0276b70e9d10b17f24f4be19d6b579810fc94eaa31783
```
[the remark this answers is not in the document]
Comment (klizhentas) [reply docs/pages/access-controls/guides/impersonation.mdx:29]: That is not necessary, they can run `tctl` anywhere

[the remark this answers is not in the document]
Comment (webvictim) [reply docs/pages/access-controls/guides/impersonation.mdx:29]: They can run `tctl status` anywhere, but creating roles (even when logged in as a user with `admin` role) fails when running remotely. I tested this just yesterday

**Pull request merged**